bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,195 @@
+                    $tag = $this->getTag();
+                    $tagsToAssign = array_filter($alreadyAssignedTags, fn ($alreadyAssignedTag) => $tag->id !== $alreadyAssignedTag->id);
+                    $tagsToAssign[] = $tag;
+
+                    $this->repository->sudo(
+                        function () use ($content, $fieldIdentifier, $tagsToAssign): void {
+                            $contentDraft = $this->repository->getContentService()->createContentDraft($content->contentInfo);
+                            $contentUpdateStruct = $this->repository->getContentService()->newContentUpdateStruct();
+
+                            $contentUpdateStruct->setField($fieldIdentifier, new TagFieldValue($tagsToAssign));
+
+                            $this->repository->getContentService()->updateContent($contentDraft->versionInfo, $contentUpdateStruct);
+                            $this->repository->getContentService()->publishVersion($contentDraft->versionInfo);
+                        }
+                    );
+
+                    $this->style->progressAdvance();
+                } catch (Throwable $t) {
+                    $this->style->error($t->getMessage());
+
+                    $this->repository->rollback();
+
+                    continue;
+                }
+            }
+            $this->repository->commit();
+        }
+
+        $this->style->progressFinish();
+        $this->style->success('Tags assigned successfully');
+
+        return Command::SUCCESS;
+    }
+
+    protected function getParentLocation(): int
+    {
+        $parentLocation = $this->input->getOption('parent-location');
+
+        if (!is_numeric($parentLocation) || (int)$parentLocation < 1) {
+            throw new InvalidOptionException(
+                sprintf("Argument --parent-location must be an integer > 0, you provided '%s'", $parentLocation),
+            );
+        }
+
+        return (int)$parentLocation;
+    }
+
+    protected function getContentTypes(): array
+    {
+        return $this->parseCommaDelimited($this->input->getOption('content-types'));
+    }
+
+    protected function getTag(): Tag
+    {
+        return $this->tagsService->loadTag($this->getTagId());
+    }
+
+    protected function getTagId(): int
+    {
+        $tagId = $this->input->getOption('tag-id');
+
+        if (!is_numeric($tagId) || (int)$tagId < 1) {
+            throw new InvalidOptionException(
+                sprintf("Argument --tag-id must be an integer > 0, you provided '%s'", $tagId),
+            );
+        }
+
+        return (int)$tagId;
+    }
+
+    private function parseCommaDelimited(?string $value): array
+    {
+        $value = trim($value ?? '');
+
+        if ($value === '') {
+            return [];
+        }
+
+        return array_values(array_unique(array_filter(array_map('trim', explode(',', $value)))));
+    }
+
+    private function hasField(Content $content, string $fieldIdentifier): bool
```

Comment:
You don't have to check like this if content has a field. The content object also has a `getField` method that returns `null` if the field does not exist.

This would eliminate unnecessary loading of content type.

Reply:
solved in e5f99dd1